src/test/java/uk/nhs/hee/tis/trainee/sync/event/ProgrammeMembershipEventListenerTest.java

```diff
@@ -54,4 +88,73 @@ void shouldCallEnricherAfterSave() {
     verify(mockEnricher).enrich(programmeMembership);
     verifyNoMoreInteractions(mockEnricher);
   }
+
+  @Test
+  void shouldFindAndCacheProgrammeMembershipIfNotInCacheBeforeDelete() {
+    ProgrammeMembership programmeMembership = new ProgrammeMembership();
+    when(cache.get(ID, ProgrammeMembership.class)).thenReturn(null);
+    when(programmeMembershipService.findById(ID.toString())).thenReturn(
+        Optional.of(programmeMembership));
+
+    Document document = new Document();
+    document.append("_id", ID);
+    BeforeDeleteEvent<ProgrammeMembership> event = new BeforeDeleteEvent<>(document, null, null);
+
+    listener.onBeforeDelete(event);
+
+    verify(cache).put(ID, programmeMembership);
+  }
+
+  @Test
+  void shouldNotFindAndCacheProgrammeMembershipIfInCacheBeforeDelete() {
+    ProgrammeMembership programmeMembership = new ProgrammeMembership();
+    when(cache.get(ID, ProgrammeMembership.class)).thenReturn(programmeMembership);
+
+    Document document = new Document();
+    document.append("_id", ID);
+    BeforeDeleteEvent<ProgrammeMembership> event = new BeforeDeleteEvent<>(document, null, null);
+
+    listener.onBeforeDelete(event);
+
+    verifyNoInteractions(programmeMembershipService);
+    verify(cache, never()).put(any(), any());
+  }
+
+  @Test
+  void shouldNotTriggerSyncWhenProgrammeMembershipNotInCacheAfterDelete() {
+    when(cache.get(ID, ProgrammeMembership.class)).thenReturn(null);
+
+    Document document = new Document();
+    document.append("_id", ID);
+    AfterDeleteEvent<ProgrammeMembership> event = new AfterDeleteEvent<>(document, null, null);
+
+    listener.onAfterDelete(event);
+
+    verifyNoInteractions(tcsService);
+  }
+
+  @Test
+  void shouldSyncDeleteWhenProgrammeMembershipNotInCacheAfterDelete() {
```

Comment:
copy-pasta? shouldSyncDeleteWhenProgrammeMembership **Not** InCacheAfterDelete() 😁